src/AutoMapper/Execution/DelegateFactory.cs

```diff
@@ -112,21 +112,20 @@ public static Expression IfNotNullExpression(MemberExpression member)
 
             return lambda;
         }
-
-        public LateBoundCtor CreateCtor(Type type)
+        
+        public Expression CreateCtor(Type type)
         {
             var ctor = _ctorCache.GetOrAdd(type, _generateConstructor);
             return ctor;
         }
 
-        private static LateBoundCtor GenerateConstructor(Type type)
+        private static Expression GenerateConstructor(Type type)
         {
             //handle valuetypes
             if(!type.IsClass())
             {
-                var ctorExpression =
-                    Expression.Lambda<LateBoundCtor>(Expression.Convert(Expression.New(type), typeof(object)));
```

Comment:
Needs the convert to object because it's a struct.  If used generics then convert wouldn't have to be called.